modules/packages/Python.chpl

```diff
@@ -2039,8 +2079,136 @@ module Python {
     }
   }
 
+  /*
+    Represents a Python tuple.  This provides a Chapel interface to Python
+    tuples, where the Python interpreter owns the tuple.
+  */
+  class PyTuple: Value {
+    @chpldoc.nodoc
+    proc init(in interpreter: borrowed Interpreter,
+              in obj: PyObjectPtr, isOwned: bool = true) {
+      super.init(interpreter, obj, isOwned=isOwned);
+    }
+
+    /*
+      Get the size of the tuple. Equivalent to calling ``len(obj)`` in Python.
+
+      :returns: The size of the tuple.
+    */
+    proc size: int throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var size = PyTuple_Size(this.getPyObject());
+      this.interpreter.checkException();
+      return size;
+    }
+
+    /*
+      Get an item from the tuple. Equivalent to calling ``obj[idx]`` in Python.
+
+      :arg T: The Chapel type of the item to return.
+      :arg idx: The index of the item to get.
+      :returns: The item at the given index.
+    */
+    pragma "docs only"
+    proc get(type T = owned Value, idx: int): T throws do
+      compilerError("docs only");
+
+    @chpldoc.nodoc
+    proc get(type T, idx: int): T throws {
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var item = PyTuple_GetItem(this.getPyObject(), idx.safeCast(Py_ssize_t));
+      this.interpreter.checkException();
+      Py_INCREF(item);
+      return interpreter.fromPythonInner(T, item);
+    }
+
+    @chpldoc.nodoc
+    proc get(idx: int): owned Value throws do
+      return this.get(owned Value, idx);
+
+    /*
+      Get a slice of the tuple indicated by ``bounds``.  Equivalent to
+      calling ``obj[bounds.low:bounds.high+1]`` in Python.
+
+      .. note::
+
+         This method does not support strided ranges or ranges with an alignment
+         other than 0.
+
+      :arg T: the Chapel type of the slice to return.
+      :arg bounds: The full slice of the tuple to return
+      :returns: A slice of the tuple for the given bounds.
+    */
+    pragma "docs only"
+    proc get(type T = owned Value, bounds: range(?)): T throws do
+      compilerError("docs only");
+
+    @chpldoc.nodoc
+    proc get(type T, bounds: range(?)): T throws {
+      if (bounds.strides != strideKind.one) {
+        compilerError("cannot call `get()` on a Python tuple with a range with stride other than 1");
```

Comment:
thinking about it, can we write `bounds: range(int, ?, strideKind.one)`?

Reply:
Are you thinking for this version of `get` or the documented version?

Reply:
All versions. Since strideKind is apart of the type, being able to specify that in the type of the range itself seems attractive

Reply:
I think doing so causes the call to become promoted over `get` instead:
```
[Executing diff myTupleBadRange2.good myTupleBadRange2.exec.out.tmp]
1,2c1
< myTupleBadRange2.chpl:8: In function 'main':
< myTupleBadRange2.chpl:16: error: cannot call `get()` on a Python tuple with a range with stride other than 1
---
> a d
[Error matching program output for library/packages/Python/correctness/myTupleBadRange2]
```

Reply:
That's unfortunate. I suppose the solution to that is to add an error overload. But then that means two error overloads. Not sure its worth it at that point.

Reply:
Although, maybe that's attractive to allow? It feels very Chapeltastic. But also probably very surprising to users

Reply:
I think I'd prefer an explicit error message over accidental promotion.  And then it feels like a toss up between whether the error is in separate overloads or in the original functions.

Reply:
I think I slightly prefer the error in separate overloads. But it really amounts to the same thing, so I leave it to you